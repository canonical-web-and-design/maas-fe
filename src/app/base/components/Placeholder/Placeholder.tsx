import type { ReactNode } from "react";

import classNames from "classnames";

type Props = {
  children: ReactNode;
  className?: string;
  loading?: boolean;
};

const Placeholder = ({
  children,
  className,
  loading = true,
}: Props): JSX.Element => {
  const delay = Math.floor(Math.random() * 750);
  if (loading) {
    return (
      <span
        aria-hidden={true}
        className={classNames("p-placeholder", className)}
        data-testid="placeholder"
        style={{ animationDelay: `${delay}ms` }}
      >
        {children}
      </span>
    );
  }
  return <>{children}</>;
};

export default Placeholder;
