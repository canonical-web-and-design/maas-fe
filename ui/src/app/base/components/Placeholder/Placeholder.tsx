import type { ReactNode } from "react";

import classNames from "classnames";

type Props = {
  children: ReactNode;
  className?: string;
};

const Placeholder = ({ children, className }: Props): JSX.Element => {
  const delay = Math.floor(Math.random() * 750);
  return (
    <span
      className={classNames("p-placeholder", className)}
      style={{ animationDelay: `${delay}ms` }}
    >
      {children}
    </span>
  );
};

export default Placeholder;
