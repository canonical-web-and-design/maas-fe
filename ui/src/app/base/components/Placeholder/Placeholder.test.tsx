import { shallow } from "enzyme";

import Placeholder from "./Placeholder";

describe("Placeholder", () => {
  beforeEach(() => {
    jest.spyOn(Math, "floor").mockReturnValue(0);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("renders", () => {
    const wrapper = shallow(<Placeholder>Placeholder text</Placeholder>);
    expect(wrapper).toMatchSnapshot();
  });
});
