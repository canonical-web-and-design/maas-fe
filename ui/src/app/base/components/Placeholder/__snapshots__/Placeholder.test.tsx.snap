// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder renders 1`] = `
<span
  className="p-placeholder"
  style={
    Object {
      "animationDelay": "0ms",
    }
  }
>
  Placeholder text
</span>
`;
